purge-tool/src/main/java/dk/dbc/holdingsitems/purge/Purge.java

```diff
@@ -0,0 +1,199 @@
+    private final boolean dryRun;
+
+    private final Connection connection;
+    private final HoldingsItemsDAO dao;
+
+    /**
+     * Create the purge request
+     * @param connection DB connection
+     * @param dao data access object for database
+     * @param worker Name of worker to put in queue
+     * @param agencyName The name of the agency to verify against
+     * @param agencyId The agency ID to be processed 
+     * @param commitEvery Optionally commit in batches when purging large sets. 0 to disable option
+     * @param dryRun Optionally check but does not commit anything
+     */
+    public Purge(Connection connection, HoldingsItemsDAO dao, String worker, String agencyName, int agencyId, int commitEvery, boolean dryRun) {
+        log.debug("Purge for agency ID {} with Queue: '{}'", agencyId, worker);
+        this.connection = connection;
+        this.dao = dao;
+        this.worker = worker;
+        this.agencyName = agencyName;
+        this.agencyId = agencyId;
+        this.commitEvery = commitEvery;
+        this.dryRun = dryRun;
+    }
+
+    /**
+     * Process the purge
+     * @throws HoldingsItemsException if DAO threw error
+     * @throws SQLException in case of rollback or commit error
+     */
+    public void process() throws HoldingsItemsException, SQLException {
+        log.debug("Process");
+
+        long start = System.currentTimeMillis();
+        
+        Set<String> bibliographicIds = dao.getBibliographicIds(agencyId);
+        int recordsCount = bibliographicIds.size();
+        int purgeCount = 0;
+        log.info("Found {} Bibliographic Ids for agency {}", recordsCount, agencyId);
+        statusReport(bibliographicIds);
+
+        // Confirm agency to purge
+        log.info("Agency: {}, Name: '{}'", agencyId, agencyName);
+        
+        boolean acceptable = false;
+        while (!acceptable) {
+            System.out.print("Enter Agency Name to confirm purge: ");
+            Scanner scanner = new Scanner(System.in);
+            String output = scanner.next();
+            acceptable = agencyName.equals(output);
+            log.debug("Response: {}", acceptable);
+            if (! acceptable) {
+                log.info("Error on '{}' != '{}'", output, agencyName);
+            }
+        }
+        log.info("Purging {}: '{}' with queue worker: '{}'", agencyId, agencyName, worker);
+        
+        purgeCount = purge(bibliographicIds);
+
+        long end = System.currentTimeMillis();
+        long duration = (end - start)/1000;
+        log.info( "Purged {} with {} live records in {} s.", recordsCount, purgeCount, duration );
+        
+        statusReport(bibliographicIds);
+    }
+
+    /**
+     * Print list statistics for agency. E.g. before and after purging
+     * @param bibliographicIds IDs to be processed
+     * @throws HoldingsItemsException if DAO threw error
+     */
+    private void statusReport(Set<String> bibliographicIds) throws HoldingsItemsException {
```

Comment:
The statusreport can be disabled by setting loglevel, which might not be obvious to the user.